### ISSUE TITLE
Use doctest to test the documentation

### ISSUE THREAD

**Issue by Hywan** [wasmerio/wasmer-python#223]
https://docs.python.org/3/library/doctest.html

**Comment by chmp** [comment]
Out of interest, I did have a quick look into this. I fear standard doctest does not find any tests as the standard syntax used in the wasmer docs is not follwing the expected repl format with lines prefixed by `>>> `:

```python
>>> import doctest
>>> import wasmer
>>> doctest.testmod(wasmer)
TestResults(failed=0, attempted=0)
```

In addition to that, pytest seems to have [troubles with native modules](https://github.com/pytest-dev/pytest/issues/1498) in its doctest integration. However, I expect it would also not find any tests, as they merely wrap the doctest module. For reference this is what you get when running pytest:

```bash
$ python -m pytest --pyargs wasmer --doctest-modules
[...]
ERROR: not found: /wasmer-python/.env/lib/python3.8/site-packages/wasmer.cpython-38-x86_64-linux-gnu.so
(no name '/wasmer-python/.env/lib/python3.8/site-packages/wasmer.cpython-38-x86_64-linux-gnu.so' in any of [])
```

You could go the way of writing a small pytest plugin, but that would the project with burden of maintining this plugin. In principle it's not overly tricky as pytest allows you to generate tests at collection time.

**Comment by chmp** [comment]
I did hack sth. together and already found issues in the docs from recent buffer PR :/ I will fix these issues in the other branch and clean up my hack in a new PR.

**Comment by Hywan** [comment]
Closed by #500.
